integration_tests/src/main/python/cache_test.py

```diff
@@ -348,3 +348,8 @@ def do_it(spark):
 def test_inmem_cache_count():
     conf={"spark.sql.session.timeZone": "America/Los_Angeles"}
     function_to_test_on_cached_df(with_gpu_session, lambda df: df.selectExpr("cast(a as timestamp)").cache().count(), int_gen, test_conf=conf)
+
+
+@pytest.mark.parametrize('with_x_session', [with_gpu_session, with_cpu_session])
+def test_batch_no_cols(with_x_session):
+    function_to_test_on_cached_df(with_x_session, lambda df: df.drop("a").cache().count(), int_gen, test_conf={})
```

Comment:
That's a valid question, the reason why it was passing is because PCBS was being used on the CPU and GPU path and since PCBS was consistently wrong on both the paths, the test passed. 
The right way to test PCBS would be to test against `DefaultCachedBatchSerializer` but it becomes difficult because `spark.sql.cache.serializer` is a static conf. It may be possible to work around it but I haven't looked into it but this highlights a valid problem and I am going to create a follow-on issue so we can come back to it at a later time.